pwiz_tools/Skyline/Model/DllFinder.cs

```diff
@@ -0,0 +1,322 @@
+                return GetSoftwareInfo(tngKey);
+            }
+            catch (Exception x)
+            {
+                return new SoftwareInfo { FailureReason = x.ToString() };
+            }
+        }
+
+        private SoftwareInfo GetSoftwareInfo(IDisposable tngKey)
+        {
+            var info = new SoftwareInfo();
+            if (tngKey == null)
+            {
+                info.FailureReason = string.Format(ModelResources.ThermoDllFinder_GetSoftwareInfo_The_key__0__was_not_found_in_the_Windows_registry_, ROOT_KEY_PATH);
+                return info;
+            }
+
+            var listMachineNames = new List<string>();
+            foreach (var subKeyName in Services.GetSubKeyNames(tngKey))
+            {
+                if (Equals(subKeyName, @"DataAccess"))
+                    continue;
+
+                listMachineNames.Add(subKeyName);
+                using var machineKey = Services.OpenSubKey(tngKey, subKeyName);
+                double? version;
+                using var versionKey = GetVersionSubKey(machineKey, out version);
+                var keyPath = GetMachineProgramPath(versionKey);
+                if (keyPath == null)
+                    continue;
+
+                // If the path for the key does not contain the necessary DLLs keep looking
+                if (!ContainsDependencyLibraries(keyPath.Path))
+                {
+                    // If no other path found yet, note the missing DLLs as the failure reason
+                    if (info.Path == null)
+                    {
+                        info.FailureReason = TextUtil.LineSeparate(
+                            string.Format(ModelResources.ThermoDllFinder_GetSoftwareInfo_The_ProgramPath__0__for_the_instrument__1__is_missing_required_files_,
+                                keyPath.Path, subKeyName),
+                            TextUtil.LineSeparate(MissingDependencyLibraries(keyPath.Path)));
+                    }
+                    continue;
+                }
+                var infoForKey = new SoftwareInfo
+                    { Path = keyPath.Path, InstrumentType = subKeyName, Version = version.Value };
+                // Return the first full installation with the DLLs
+                if (keyPath.IsFull)
+                    return infoForKey;
+                // Otherwise, if the current best path is null, use this one
+                if (info.Path == null)
+                    info = infoForKey;
+            }
+
+            if (info.Path == null && info.FailureReason == null)
+            {
+                info.FailureReason = TextUtil.LineSeparate(string.Format(ModelResources.ThermoDllFinder_GetSoftwareInfo_Failed_to_find_a_machine_name_key_with_a_valid_ProgramPath, ROOT_KEY_PATH),
+                    TextUtil.LineSeparate(listMachineNames));
+            }
+
+            return info;
+        }
+
+        private class ProgramPath
+        {
+            public string Path;
+            public bool IsFull;
```

Comment:
This probably should be called "IsFullInstall" because otherwise it sure sounds like it means "IsFullPath" given the class it's in.

Reply:
Okay. Changed to IsFullInstall and added comments.